website/blog/2023-09-25-preparing-your-site-for-docusaurus-v3/index.mdx

```diff
@@ -0,0 +1,527 @@
+
+Each MDX version comes with its own playground:
+
+- [MDX v1 playground](https://mdx-git-renovate-babel-monorepo-mdx.vercel.app/playground/)
+- [MDX v2 playground](https://mdxjs.com/playground/)
+
+<details>
+  <summary>Configure the [MDX v2 playground](https://mdxjs.com/playground/) options.</summary>
+
+To obtain a compilation behavior similar to what Docusaurus v2 uses, please turn on these options on the [MDX v2 playground](https://mdxjs.com/playground/):
+
+- Use `MDX`
+- Use `remark-gfm`
+- Use `remark-directive`
+
+![Configure MDX v2 playground options for Docusaurus v3 behavior](./img/mdx2-playground-options.png)
+
+</details>
+
+Using the 2 MDX playgrounds side-by-side, you will soon notice that some content is compiled differently, or fails to compile in v2. We will detail the most common cases below.
+
+:::tip Making your content future-proof
+
+The goal will be to refactor your problematic content so that it **works fine with both versions of MDX**. This way, when you upgrade to Docusaurus v3, this content will already work out-of-the-box.
+
+:::
+
+### Using the MDX checker CLI
+
+We provide a [docusaurus-mdx-checker](https://github.com/slorber/docusaurus-mdx-checker) CLI that permits to easily spot problematic content. Run this command today on your Docusaurus v2 site to obtain a list of files that will fail to compile under MDX v2.
+
+```bash
+npx docusaurus-mdx-checker
+```
+
+For each compilation issue, the CLI will log the file path and a line number to look at.
+
+![MDX Checker CLI output](./img/mdx-checker-output.png)
+
+:::tip
+
+Use this CLI to estimate of how much work will be required to make your content compatible with MDX v2.
+
+:::
+
+:::warning
+
+This CLI is a best effort, and will only report compilation errors.
+
+It will not report subtle compilation changes that do not produce errors but can affect how your content is displayed. To catch these problems, we recommend using [visual regression tests](/blog/upgrading-frontend-dependencies-with-confidence-using-visual-regression-testing).
+
+:::
+
+### Common MDX problems
+
+We upgraded a few Docusaurus sites to Docusaurus v3 and MDX v2:
+
+- [Docusaurus PR](https://github.com/facebook/docusaurus/pull/8288)
+- [React-Native PR](https://github.com/facebook/react-native-website/pull/3780)
+- [Jest PR](https://github.com/jestjs/jest/pull/14463)
+
+These upgrades permitted us to aggregate the most common content problems, and document how to best handle them.
+
+#### Bad usage of `{`
+
+The `{` character is used for opening [JavaScript expressions](https://mdxjs.com/docs/what-is-mdx/#expressions). MDX v2 will now fail if what you put inside `{expression}` is not a valid expression.
+
+```md title="example.md"
+The object shape looks like {username: string, age: number}
+```
+
+:::danger Error message
+
+> Could not parse expression with acorn: Unexpected content after expression
+
+:::
+
+:::tip How to prepare
```

Comment:
Agree, but for consistency with the structure of all the subsequent section I thought this would be easier to scan for preparatory instructions visually.

If you have a better idea than admonition that would keep the ability to quickly scan and find instructions, feel free to send a PR after publication